Remove and replace WKBundlePageSetTextZoomFactor/SetPageZoomFactor

charliewolfe opened the pull request:
#### a42f5c20a731b7d19a60187b3330cfb916dca7d9
<pre>
Remove and replace WKBundlePageSetTextZoomFactor/SetPageZoomFactor
<a href="https://rdar.apple.com/126969249">rdar://126969249</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=273170">https://bugs.webkit.org/show_bug.cgi?id=273170</a>

Reviewed by Sihui Liu.

Replace these functions with a synchronous message to the UI process. This is required because we will
need to have the UI process zoom all web processes with site isolation.

* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
(WKBundlePageSetTextZoomFactor): Deleted.
(WKBundlePageSetPageZoomFactor): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePagePrivate.h:
* Tools/WebKitTestRunner/InjectedBundle/EventSendingController.cpp:
(WTR::EventSendingController::textZoomIn):
(WTR::EventSendingController::textZoomOut):
(WTR::EventSendingController::zoomPageIn):
(WTR::EventSendingController::zoomPageOut):
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::InjectedBundlePage::prepare):
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::didReceiveSynchronousMessageFromInjectedBundle):
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::invoke):

Canonical link: <a href="https://commits.webkit.org/277991@main">https://commits.webkit.org/277991@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/12da23e6138553d14ee3712205b098789a73a2a0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/49089 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28331 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/52085 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51836 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/45171 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/51393 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34301 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25865 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/40113 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/50595 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25931 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42314 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21227 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/23386 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/43490 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/7296 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/45318 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43993 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53751 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/24165 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20371 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/47433 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/25445 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42520 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/46402 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26233 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7046 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/25164 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai reviewed (commented):
From the commit message, it's not very clear why this change is needed for site isolation.
Is the problem that with site isolation, the web process that hosts the bundle page may not be the main frame process and the factor needs to be set in the main frame process?

achristensen07 commented on Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
```diff
@@ -398,21 +398,11 @@ double WKBundlePageGetTextZoomFactor(WKBundlePageRef pageRef)
-    WebKit::toImpl(pageRef)->setTextZoomFactor(zoomFactor);
-}
-
 double WKBundlePageGetPageZoomFactor(WKBundlePageRef pageRef)
```
Let's remove these accessors too if possible.

charliewolfe replied:
I can do this, but it doesn’t seem necessary to send another IPC message to the UI process to get the current zoom. The zoom factor should be the same in the main frame process, and we need this API to get that.

szewai commented on Tools/WebKitTestRunner/InjectedBundle/EventSendingController.cpp:
```diff
@@ -461,41 +461,57 @@ JSValueRef EventSendingController::contextClick()
+    postSynchronousPageMessage("EventSender", body);
+    setValue(body, "SubMessage", "ZoomText");
+    setValue(body, "ZoomFactor", zoomFactor);
+    postSynchronousPageMessage("EventSender", body);
```
It seems unnecessary to send two sync messages; we could add `EventSenderProxy::textZoomIn` (etc) and let UI process figure out what to do.

charliewolfe committed:
Remove and replace WKBundlePageSetTextZoomFactor/SetPageZoomFactor
rdar://126969249
https://bugs.webkit.org/show_bug.cgi?id=273170

Reviewed by Sihui Liu.

Replace these functions with a synchronous message to the UI process. This is required because we will
need to have the UI process zoom all web processes with site isolation.

* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
(WKBundlePageSetTextZoomFactor): Deleted.
(WKBundlePageSetPageZoomFactor): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePagePrivate.h:
* Tools/WebKitTestRunner/InjectedBundle/EventSendingController.cpp:
(WTR::EventSendingController::textZoomIn):
(WTR::EventSendingController::textZoomOut):
(WTR::EventSendingController::zoomPageIn):
(WTR::EventSendingController::zoomPageOut):
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::InjectedBundlePage::prepare):
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::didReceiveSynchronousMessageFromInjectedBundle):
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::invoke):

Canonical link: https://commits.webkit.org/277991@main
